### DISCUSSION TITLE
Error: module 'MumbleServer' has no attribute 'MetaPrx'

### DISCUSSION THREAD

**Discussion by FrankRulof** [zeroc-ice/ice#1980] [Q&A] [answered]
this error is on the following python 3.12 code 
import Ice
import MumbleServer_ice 
import sys

def main():
    with Ice.initialize(sys.argv) as communicator:
        proxy = communicator.stringToProxy("Meta:tcp -h 192.168.1.245 -p 6502")
        try:
            meta = MumbleServer_ice.MetaPrx.checkedCast(proxy)
            if not meta:
                raise RuntimeError("Invalid proxy")
            
        except Exception as e:
            print("Error:", e)
        
        #meta = Murmur.MetaPrx.checkedCast(proxy)

        # Authenticate (if required)
        # ...

        # Perform other Murmur-related actions
        # ...

if __name__ == "__main__":
    main()

try:
    meta = MumbleServer_ice.MetaPrx.checkedCast(proxy)
    if not meta:
        raise RuntimeError("Invalid proxy")
except Exception as e:
    print("Error:", e)

I made this sample program to get the point the py file was generated with slice2py. I added both the ICE and the Py files
As far as i can see the class MetPrx is in the MumbleServer_ice.py file , so i dont understand were the problem is. I hope the forum can help. THank you all in advance

**Comment by FrankRulof** [comment]
the error i get is Error: module 'MumbleServer' has no attribute 'MetaPrx'

**Comment by pepone** [comment, marked as answer]
Hi Frank,

You should not import `import MumbleServer_ice` directly, the generated code includes a Module in `MumbleServer/__init__.py`, which you need to import.

Something like:

```
import Ice
import MumbleServer


with Ice.initialize() as communicator:
    meta = MumbleServer.MetaPrx.checkedCast(
        communicator.stringToProxy("Meta:tcp -h 192.168.1.245 -p 6502")
    )
    if not meta:
        raise RuntimeError("Invalid proxy")
```

**Comment by FrankRulof** [comment]
thank you very much , it did the job
